tools/discovery/discovery.go

```diff
@@ -89,6 +89,10 @@ func refEnginesFetching(parsedName map[string]string) (uri *url.URL, engines []e
 		return uri, nil, fmt.Errorf("ref engine fetching error, status code = %d", resp.StatusCode)
 	}
 
+	if resp.Header.Get(`Content-Type`) != `application/vnd.oci.ref-engines.v1+json; charset=utf-8` {
```

Comment:
We can accept other charsets, although Go's JSON parser may be limited to [these][1].  Maybe splo=it on the optipnal semicolon to remove the charset portion (if present).

[1]: https://tools.ietf.org/html/rfc7159#section-8.1

Reply:
> Maybe split on the optional semicolon…

The Go for this seems to be [`ParseMediaType`][1].

[1]: https://golang.org/pkg/mime/#ParseMediaType